scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -455,3 +457,38 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
         return relativePathToScannedFile.invariantSeparatorsPath
     }
 }
+
+/**
+ * Work around to prevent that duplicate [ScanResult]s from the [ScanResultsStorage] do get duplicated in the
+ * [OrtResult] produced by this scanner.
+ *
+ * The time interval between a failing look-up of the cache entry and the resulting scan with the following store
+ * operation can be relatively large. Thus this [LocalScanner] is prone to adding duplicate scan results if scans
+ * are run in parallel. In particular the [PostgresStorage] allows adding duplicate tuples
+ * (identifier, provenance, scanner details) which probably should be made unique.
+ *
+ * TODO:
+ *
+ * 1. Minimize the time between the failing look-up and the curresponding store operation mentioned.
+ * 2. Make the tuples (identifier, provenance, scanner details) unique (dis-regarding provenance.downloadTime), at
+ * least in [PostgresStroage].
+  */
+private fun ScanResultContainer.deduplicateScanResults(): ScanResultContainer {
+    // Use vcsInfo and sourceArtifact instead of provenance in order to ingore the download time and original VCS info.
+    data class Key(
+        val id: Identifier,
+        val vcsInfo: VcsInfo?,
+        val sourceArtifact: RemoteArtifact?,
+        val scannerDetails: ScannerDetails
+    )
+    fun ScanResult.key() = Key(id, provenance.vcsInfo, provenance.sourceArtifact, scanner)
+
+    val deduplicatedResults = results.associateBy { it.key() }.values.toList()
```

Comment:
How about `results.distinctBy { it.key() }`?

Reply:
Done.